### ISSUE TITLE
6.0.0 regression: GenerateMetadataError: Unknown type: NeverKeyword

### ISSUE THREAD

**Issue by hpx7** [lukeautry/tsoa#1535]
## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

In 5.1.1, tsoa was able to handle the following type (which represents an empty object, i.e. `{}`)

```ts
Record<string, never>
```

## Current Behavior

This breaks starting in 6.0.0-rc.0

```
There was a problem resolving type of 'Record<string, never>'.

Generate routes error.
 GenerateMetadataError: Unknown type: NeverKeyword
At unknown position...
This was caused by '<unknown name>'
    at TypeResolver.resolve (/project/node_modules/@tsoa/cli/src/metadataGeneration/typeResolver.ts:470:13)
    at <anonymous> (/project/node_modules/@tsoa/cli/src/metadataGeneration/typeResolver.ts:244:113)
```

## Steps to Reproduce

1. 2. 3. 4.

## Context (Environment)

Version of the library: 6.0.0
Version of NodeJS: 18.17.1

- Confirm you were using yarn not npm: [ ]

## Breaking change?

This is a breaking change in tsoa version 6

**Comment by WoH** [comment]
This was a bug in the old implementation, but we should support this.

**Comment by hpx7** [comment]
@WoH I'm happy to try and contribute support for this, would you be able to provide any guidance for how this should be done?

**Comment by WoH** [comment]
This should result in:

```json
{
  "type": "object",
  "properties": {
  },
  "additionalProperties": false
}
```

Right?

**Comment by hpx7** [comment]
This is what it generates on 5.1.1:
```json
	"Record_string.never_": {
		"properties": {},
		"type": "object",
		"description": "Construct a type with a set of properties K of type T"
	},
```

**Comment by WoH** [comment]
Since we are now trying to descend into the indexed type in a Mapped Type, we need to probably try to check if the indexType is the never type and handle that separately: 
https://github.com/lukeautry/tsoa/blob/master/packages/cli/src/metadataGeneration/typeResolver.ts#L242

**Comment by hpx7** [comment]
Thanks for the tip, I created https://github.com/lukeautry/tsoa/pull/1537

**Comment by WoH** [comment]
Resolved in v6.0.1
